src/plugins/unified_histogram/public/layout/layout.tsx

```diff
@@ -263,17 +269,16 @@ export const UnifiedHistogramLayout = ({
 
   const currentTopPanelHeight = topPanelHeight ?? defaultTopPanelHeight;
 
-  const onResetChartHeight = useMemo(() => {
-    return currentTopPanelHeight !== defaultTopPanelHeight &&
-      panelsMode === ResizableLayoutMode.Resizable
-      ? () => onTopPanelHeightChange?.(undefined)
-      : undefined;
-  }, [currentTopPanelHeight, defaultTopPanelHeight, onTopPanelHeightChange, panelsMode]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const chartMemoized = useMemo(() => chart, [chart?.title, chart?.timeInterval, chart?.hidden]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const hitsMemoized = useMemo(() => hits, [hits?.status, hits?.total]);
```

Comment:
Just curious if the lack of memoization before was causing any issues in this PR, or if this was just added for convenience to prevent consumers having to do their own memoization? It's not an issue to do it here IMO (although I don't love having to disable ESLint rules personally), I'm just wondering. In retrospect, given how Unified Histogram has evolved I probably should have done this differently from the start and skipped the context objects altogether (I guess hindsight is 20/20).

Reply:
@davismcphee That was something I added when trying to reduce the number of Chart rerenderings. I can't prove now that it's still relevant so I'm removing this code.